File: 프로그래머스/1/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.py
def solution(bandage, health, attacks):
    
    cur_h = health
    before_t = 0
    
    for attack in attacks:
        
        gap = attack[0] - before_t - 1
        
        # 현재 체력 세팅
        cur_h = min(cur_h + (gap * bandage[1]) + (gap // bandage[0]) * bandage[2], health)

        # 괴수의 어택
        cur_h = cur_h - attack[1]

        if cur_h <= 0:
            return -1
        
        before_t = attack[0]
    
    return cur_h

File: 프로그래머스/1/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.java
class Solution {
    public int solution(int[] bandage, int health, int[][] attacks) {
        
        int curHealth = health;
        int beforeTime = 0;
        
        for (int[] attack : attacks) {
            int gap = attack[0] - beforeTime - 1;
            
            // 현재 체력 회복
            curHealth = Math.min(curHealth + gap * bandage[1] + (gap / bandage[0]) * bandage[2], health);
                
            // 몬스터 공격
            curHealth -= attack[1];
                
            if (curHealth <= 0) {
                return -1;
            };
            
            beforeTime = attack[0];
        }
        
        return curHealth;
    }
}